app/src/test/java/org/glucosio/android/tools/GlucoseRangesTest.java

```diff
@@ -0,0 +1,191 @@
+package org.glucosio.android.tools;
+
+
+import android.content.Context;
+import android.support.v4.content.ContextCompat;
+
+import org.glucosio.android.BuildConfig;
+import org.glucosio.android.R;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.RuntimeEnvironment;
+import org.robolectric.annotation.Config;
+
+import static junit.framework.Assert.assertEquals;
+import static junit.framework.Assert.assertFalse;
+import static junit.framework.Assert.assertTrue;
+
+@RunWith(RobolectricTestRunner.class)
+@Config(constants = BuildConfig.class, packageName = "org.glucosio.android", sdk = 19)
+public class GlucoseRangesTest {
+
+    private final Context contextMock = RuntimeEnvironment.application.getApplicationContext();
+    private TestGlucoseRanges glucoseRanges = new TestGlucoseRanges(contextMock);
+    private int reading = 65;
+
+    @Test
+    public void expectIsInUnitTestsReturnTrue() {
+        assertTrue(glucoseRanges.isInUnitTests());
+    }
+
+    @Test
+    public void expectIsInUnitTestsReturnFalseByDefault() {
+        assertFalse(new GlucoseRanges().isInUnitTests());
+    }
+
+    @Test
+    public void expectPurpleWhenReadingValueIs65WithNoPreferredRange() {
+        assertTrue(glucoseRanges.colorFromReading(reading).equals("purple"));
+    }
+
+    @Test
+    public void expectRedWhenReadingValueIs210WithNoPreferredRange() {
+        reading = 410;
```

Comment:
this is quite strange to have the field that has the default value and it is changed in almost every test method. I would just to have a local variable in every test.

Reply:
I thought that using local variable for every test method would be a nice example of duplicating code. I would use local variable in test method only when it would be used only there.

My idea was simply to create one variable and reuse it as much I can.

Reply:
That is why I suggested you to write parametrized test. Which is a bit cumbersome code with junit4.